blocks/rich-text/index.js

```diff
@@ -683,28 +689,20 @@ export class RichText extends Component {
 		this.setState( { formats, focusPosition, selectedNodeId: this.state.selectedNodeId + 1 } );
 	}
 
-	updateContent() {
-		const bookmark = this.editor.selection.getBookmark( 2, true );
-		this.savedContent = this.props.value;
-		this.setContent( this.savedContent );
-		this.editor.selection.moveToBookmark( bookmark );
-
-		// Saving the editor on updates avoid unecessary onChanges calls
-		// These calls can make the focus jump
-		this.editor.save();
-	}
-
 	setContent( content = '' ) {
 		this.editor.setContent( renderToString( content ) );
 	}
 
 	getContent() {
+		if ( this.state.empty ) {
```

Comment:
In #4953 I discovered that sometimes the `getContent`/`onChange` is called before the `this.state.empty` is updated. I think we should not rely on it here and recompute the `empty` from the content. I think it's performant enough.

I was also wondering if we chould update this state as part of the `onChange` handler (instead of selectionChange) but could be left as a separate enhancement.

Reply:
Yeah, this is a good point. Changes through `setState` are not guaranteed to be applied immediately, so either we need to save it as a class property or just simply add it here. I think the latter sounds great as we can then also remove the selection change event. I can't think of a case where input wouldn't be fired when the field is emptied or filled?

Reply:
> I can't think of a case where input wouldn't be fired when the field is emptied or filled?

Well I discovered that if you make a selection (a word or sentense) and click backspace, it's not triggered. This should be fixed anyway because it's creating a small bug in the side inserter. I was thinking at maybe calling `onChange` on `keydown` or something?

Reply:
Yeah, that's why the keyup handler is there, because `selectionChange` has the same issue, at least in Chrome. I updated this PR. Nice small performance improvement as well because `selectionChange` fires waaay more often.